### PR TITLE
flask: add request fitler

### LINKED ISSUE #106: Extract flask request processing code into notifier filter
Code in https://github.com/airbrake/pybrake/blob/master/pybrake/flask.py#L96-L125 should be a filter

### PR DESCRIPTION
**Pull request by anmic**
Fixes #106.

### CODE REVIEW

**Comment by anmic** [comment]
@vmihailenco PTAL

File: pybrake/flask.py [hunk 1]
```diff
@@ -37,6 +37,9 @@ def init_app(app):
     notifier = Notifier(**app.config["PYBRAKE"])
     apm_disabled = notifier._apm_disabled
 
+    if not apm_disabled:
```
Comment (vmihailenco) [inline pybrake/flask.py]: what this check achieves?

**Comment by vmihailenco** [comment]
Did you check that this code works?

**Comment by anmic** [comment]
Fixed request filter.

**Comment by vmihailenco** [comment]
What was broken and what is the fix.

**Comment by anmic** [comment]
Before that, I saved the request in thread-local storage. `_handle_exception` and `request_filter` functions are processed in different threads. And It did not work.
Now I save environ in notice context because I don't know how else I can pass the request to `request_filter`. 
Then I get the request context using environ.

**Comment by vmihailenco** [comment]
In separate PR change `send_notice` to run filters in the same thread before passing them to `send_notice_sync` (make sure that filters are applied once). Then change this PR to work as before.

**Comment by anmic** [comment]
I saved the request object in thread-local storage and made rebase on master.

**Comment by vmihailenco** [comment]
Flask already does that for you - https://flask.palletsprojects.com/en/1.1.x/quickstart/#accessing-request-data. No need to do it twice.

**Comment by anmic** [comment]
I use the Flask request object in the request filter.

File: pybrake/flask.py [hunk 2]
```diff
@@ -28,6 +28,41 @@
 _UNKNOWN_ROUTE = "UNKNOWN"
 
 
+def request_filter(notice):
+    ctx = notice["context"]
+    ctx["method"] = request.method
```
Comment (vmihailenco) [inline pybrake/flask.py]: And what will happen if this is called when request is not available?

Comment (anmic) [reply pybrake/flask.py]: Added check if the request object is not available.

**Pull request merged**